pyproject.toml

```diff
@@ -6,25 +6,11 @@ requires = [
     "Cython>=0.28.5",
 
     # PyPy needs numpy >= 1.14.0
-    # platform_python_implementation!='CPython' not needed >= Python 3.7 which is numpy 1.14.5
-    "numpy==1.13.3; python_version=='3.6' and platform_machine!='aarch64' and platform_system!='AIX' and platform_python_implementation=='CPython'",
-    "numpy==1.14.0; python_version=='3.6' and platform_machine!='aarch64' and platform_system!='AIX' and platform_python_implementation!='CPython'",
+    "numpy==1.14.0; python_version=='3.6' and platform_machine!='aarch64' and platform_system!='AIX' and platform_python_implementation=='PyPy'",
```

Comment:
I updated this to specifically call out PyPy since this is what PyPy reports.

Reply:
In #18879, I have realized that `numpy==1.19.0` has to be used for `PyPy` (the wheels are available from this version). Otherwise, `numpy` is compiled from source and this is something that we should avoid.

WDYT?

Reply:
I think this change doesn't change anything there so it's not problematic immediately (not a regression and might not be something that should block this PR). I do think that if I were to bump this here I'd need to also add your change here https://github.com/scikit-learn/scikit-learn/pull/18879/files#diff-83a4b90ed0c61f5e1a834555270a9d1e687cd4ece074262fcd18e7fcf160e6d5 and it might make sense to bump the version in #18879 unless you think that will merge before this (and I could update the change to match `master`)

Reply:
PyPy support PR for `oldest-supported-numpy`: https://github.com/scipy/oldest-supported-numpy/pull/15

> I have realized that `numpy==1.19.0` has to be used for `PyPy` (the wheels are available from this version). Otherwise, `numpy` is compiled from source and this is something that we should avoid.

That sounds right to me.

Reply:
@alfaro96 did you want me to bump the min supported numpy for PyPy to be 1.19.0? As @rgommers mentions the PyPy spec of 1.19.0 has been upstreamed to `oldest-supported-numpy`. If we're going to be bumping the version here I might as well simplify this all to just use that with no constraints.

Reply:
I think that we could rely on `oldest-supported-numpy` for `PyPy`. Then we would not need to keep track of the `NumPy` wheels available for each architecture (as @rth stated in https://github.com/scikit-learn/scikit-learn/pull/18900#issuecomment-733689140).

Reply:
I think that we need to update the minimum version of `NumPy` for `PyPy` in `sklearn/_min_dependecies.py`.

CC @thomasjpfan.

Reply:
> Moreover, I think that we need to update the minimum version of `NumPy` for `PyPy` in `sklearn/_min_dependencies.py`.
> 
> CC @thomasjpfan.

That's actually what I meant about bumping the min supported numpy :sweat_smile: